hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java

```diff
@@ -277,6 +277,7 @@ public static boolean isReadOnly(
       // Although indirectly the Ranger sync service task could invoke write
       // operation SetRangerServiceVersion.
     case GetKeyInfo:
+    case TransferLeadership:
```

Comment:
I think we should position TransferLeadership as write operation instead of read operation for it's actually change the state of the cluster.

Reply:
If it's a write operation, the request would be sent to all 3 OMs, not just the leader one.
Besides, I think the the request is also treated as a READ operation in ratis? https://github.com/apache/ratis/blob/master/ratis-common/src/main/java/org/apache/ratis/protocol/TransferLeadershipRequest.java#L25

Reply:
Just to clarify: The transferLeadershipRequest is a read request while the setConfigurationRequest is a write request (From Ratis code).

I think the reason why we should still keep it as a READ operation for OM @ChenSammi 
1. Just like @symious said, the write request would be sent to all 3 OMs
2. The transfer Leadership process will not write to OM's DB, the state change is on the Ratis level, whose action is guaranteed by the ratis. Could refer to **RangerBGSync** above, which also does not write to OM's db.

Reply:
+1 transfer leadership should be a "read" request, since our definition of read is anything that does not modify the state machine (RocksDB).

Reply:
@Xushaohong, thanks for the explain.  It makes sense to keep it as read.